### ISSUE TITLE
Possibilitar desligamento por desistência durante período de trancamento de matrícula

### ISSUE THREAD

**Issue by helioaga** [gems-uff/sapos#479]
Atualmente quando há tentativa de desligar por desistência um aluno no período em que a matrícula está trancada, o SAPOS não permite (tela abaixo).

![image](https://github.com/gems-uff/sapos/assets/170034455/6447d7be-6292-44d1-ac2a-031fe2c492a3)

Após conversar com Coordenação do Curso, fui orientado a criar esta issue solicitando a verificação e configuração do sistema para que o desligamento por desistência seja possível durante o período de trancamento de matrícula.

**Comment by leomurta** [comment]
@IgorMonardez , essa aqui deve ser bem fácil de fazer. Quando vc terminar de atualizar as libs, pega essa, tá?

**Comment by SAASVUUV** [comment]
Verifiquei, com a ajuda do João, que o problema dessa issue estava relacionada ao trancamento da matrícula. No caso, ao ser feito a atualização, o trancamento é validado e é feita uma verificação na sua data. Então, se a data de término do trancamento ficava após a data de desligamento 
`if self.end_date > dismissal.date -> error`
é lançado o erro. Porém, com essa configuração, caso fosse feito o desligamento desse aluno antes do término do trancamento e após o começo desse (ou seja, dentro do período trancado) é lançado o erro, o que não faz sentido, visto que deveria ser possível desligar a matrícula durante tal ocasião. Por isso, acredito que faria sentido trocar a verificação para 
`if self.start_date > dismissal.date -> error`
Pois dessa maneira mantém-se a coerência de não poder ser feito o tracamento após o desligamento do aluno.
Após feitos os testes, vi que o bug da issue foi resolvido e foi lançado o erro quando era coerente. Posso dar um commit?

**Comment by leomurta** [comment]
@SAASVUUV acho que faz sentido a sua proposta. Por mim, pode seguir em frente. Obrigado!

**Comment by SAASVUUV** [comment]
Sim, eu vi. Fiquei um bom tempo pra entender o que estava dando o erro. Mas isso deve-se pelo fato de que eu não havia considerado o número de semestres. Então deixei de maneira que fosse possível somente fazer o desligamento no último período de trancamento, visto que o último período de trancamento deve ser o período no qual houve o desligamento. Em breve realizarei um segundo PR.

**Comment by leomurta** [comment]
@helioaga essa demanda foi implementada pelo @SAASVUUV e já está no ar. Obrigado Luis Felipe!
